app/flatpak-builtins-info.c

```diff
@@ -201,51 +216,107 @@ flatpak_builtin_info (int argc, char **argv, GCancellable *cancellable, GError *
       if (strcmp (commit, latest) != 0)
         {
-          g_print ("%s%s%s %s\n", on, _("Active commit:"), off, commit);
-          g_print ("%s%s%s %s\n", on, _("Latest commit:"), off, latest);
+          len = MAX (len, strlen (_("Active commit:")));
+          len = MAX (len, strlen (_("Latest commit:")));
         }
       else
-        g_print ("%s%s%s %s\n", on, _("Commit:"), off, commit);
+        len = MAX (len, strlen (_("Commit:")));
+      if (parent)
+        len = MAX (len, strlen (_("Parent:")));
       if (alt_id)
-        g_print ("%s%s%s %s\n", on, _("alt-id:"), off, alt_id);
-      g_print ("%s%s%s %s\n", on, _("Parent:"), off, parent ? parent : "-");
-      g_print ("%s%s%s %s\n", on, _("Location:"), off, path);
-      g_print ("%s%s%s %s\n", on, _("Installed size:"), off, formatted);
+        len = MAX (len, strlen (_("alt-id:")));
+      len = MAX (len, strlen (_("Parent:")));
       if (eol)
-        g_print ("%s%s%s %s\n", on, _("end-of-life:"), off, eol);
+        len = MAX (len, strlen (_("end-of-life:")));
       if (eol_rebase)
-        g_print ("%s%s%s %s\n", on, _("end-of-life-rebase:"), off, eol_rebase);
+        len = MAX (len, strlen (_("end-of-life-rebase:")));
+      if (strcmp (parts[0], "app") == 0)
+        len = MAX (len, strlen (_("Sdk:")));
+      if (subpaths[0] != NULL)
+        len = MAX (len, strlen (_("Subdirectories:")));
+
+      flatpak_get_window_size (&rows, &cols);
+      width = cols - (len + 1);
+
+      g_print ("%s%*s%s %s\n", on, len, _("ID:"), off, parts[1]);
+      g_print ("%s%*s%s %s\n", on, len, _("Ref:"), off, ref);
+      g_print ("%s%*s%s %s\n", on, len, _("Arch:"), off, parts[2]);
+      g_print ("%s%*s%s %s\n", on, len, _("Branch:"), off, parts[3]);
+      g_print ("%s%*s%s %s\n", on, len, _("Version:"), off, app ? as_app_get_version (app) : "");
+      g_print ("%s%*s%s %s\n", on, len, _("Origin:"), off, origin ? origin : "-");
+      if (collection_id)
+        g_print ("%s%*s%s %s\n", on, len, _("Collection:"), off, collection_id);
+      g_print ("%s%*s%s %s\n", on, len, _("Installation:"), off, flatpak_dir_get_name_cached (dir));
+      g_print ("%s%*s%s %s\n", on, len, _("Installed:"), off, formatted);
       if (strcmp (parts[0], "app") == 0)
         {
           g_autofree char *runtime = NULL;
-          g_autofree char *sdk = NULL;
           runtime = g_key_file_get_string (metakey, "Application", "runtime", error);
-          g_print ("%s%s%s %s\n", on, _("Runtime:"), off, runtime ? runtime : "-");
+          g_print ("%s%*s%s %s\n", on, len, _("Runtime:"), off, runtime ? runtime : "-");
+        }
+      g_print ("\n");
+
+      if (strcmp (parts[0], "app") == 0)
+        {
+          g_autofree char *sdk = NULL;
           sdk = g_key_file_get_string (metakey, "Application", "sdk", error);
-          g_print ("%s%s%s %s\n", on, _("Sdk:"), off, sdk ? sdk : "-");
+          g_print ("%s%*s%s %s\n", on, len, _("Sdk:"), off, sdk ? sdk : "-");
```

Comment:
Its a bit to have the Sdk after the above newline. I thought its supposed to be app data above and then commit data below?

Reply:
I wasn't sure. moving it up is fine with me